async-nats/src/lib.rs

```diff
@@ -610,23 +630,69 @@ impl Connector {
 /// Client is a `Clonable` handle to NATS connection.
 /// Client should not be created directly. Instead, one of two methods can be used:
 /// [connect] and [ConnectOptions::connect]
-#[derive(Clone)]
 pub struct Client {
     sender: mpsc::Sender<ClientOp>,
     subscription_context: Arc<Mutex<SubscriptionContext>>,
+    errors: Option<tokio::sync::mpsc::Receiver<String>>,
+}
+
+impl Clone for Client {
+    fn clone(&self) -> Self {
+        Client {
+            sender: self.sender.clone(),
+            subscription_context: self.subscription_context.clone(),
+            errors: None,
+        }
+    }
 }
 
 impl Client {
     pub(crate) fn new(
         sender: mpsc::Sender<ClientOp>,
         subscription_context: Arc<Mutex<SubscriptionContext>>,
+        errors: Option<mpsc::Receiver<String>>,
     ) -> Client {
         Client {
             sender,
             subscription_context,
+            errors,
         }
     }
 
+    /// Returns stream of asynchronous errors received from NATS server.
+    ///
+    /// # Examples
+    /// ```
+    /// # use futures_util::StreamExt;
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    /// let mut nc = async_nats::connect("demo.nats.io").await?;
+    ///
+    /// let mut errs = nc.errors_stream().await?;
+    /// tokio::spawn({
+    ///    async move {
+    ///        if let Some(err) = errs.next().await {
+    ///             println!("received error: {}", err);
+    ///        };
+    ///    }
+    /// });
+    /// # Ok(())
+    /// # }
+    ///
+    /// ```
+    pub async fn errors_stream(&mut self) -> io::Result<ErrorsStream> {
```

Comment:
@caspervonb have an idea for better name?

Reply:
Maybe just `errors` or `error_notifications`